jerry-core/ecma/builtin-objects/ecma-builtins.c

```diff
@@ -995,7 +995,18 @@ ecma_builtin_list_lazy_property_names (ecma_object_t *object_p, /**< a built-in
       {
         ecma_value_t name = ecma_make_magic_string_value ((lit_magic_string_id_t) curr_property_p->magic_string_id);
 
-        ecma_collection_push_back (for_non_enumerable_p, name);
+#if ENABLED (JERRY_ES2015)
+        if (curr_property_p->attributes & ECMA_PROPERTY_FLAG_ENUMERABLE)
+        {
+          ecma_collection_push_back (main_collection_p, name);
+        }
+        else
+        {
+#endif /* ENABLED (JERRY_ES2015) */
+          ecma_collection_push_back (for_non_enumerable_p, name);
+#if ENABLED (JERRY_ES2015)
```

Comment:
It would be good to find a way to avoid the guards for closing brackets. What if we do this? 
```c
#if ENABLED(...)
  if (...)
  {
    ...
  }
else
#endif
  {
    ..
  }
```

Still not an ideal one, but at least we have removed the need for the second #ifdef guard.
@dbatyai  What do you think?

Reply:
+1

Reply:
Good idea! I've updated the PR.